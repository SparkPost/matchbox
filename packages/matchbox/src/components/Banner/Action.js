import React from 'react';
import { Button } from '../Button';

const Action = React.forwardRef(function Action(props, userRef) {
  return <Button mt={['300', null, '500']} mr="400" {...props} ref={userRef} />;
});

Action.displayName = 'Banner.Action';
Action.propTypes = {
  ...Button.propTypes,
};

export default Action;
