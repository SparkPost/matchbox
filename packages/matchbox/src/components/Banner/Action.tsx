import React from 'react';
import { Button } from '../Button';

type ActionProps = React.ComponentProps<typeof Button>;

const Action = React.forwardRef<HTMLButtonElement, ActionProps>(function Action(
  props: ActionProps,
  userRef,
) {
  return <Button mt={['300', null, '500']} mr="400" {...props} ref={userRef} />;
}) as React.ForwardRefExoticComponent<ActionProps> & {
  Button: typeof Button;
};

Action.displayName = 'Banner.Action';
Action.propTypes = {
  ...Button.propTypes,
};

export default Action;
