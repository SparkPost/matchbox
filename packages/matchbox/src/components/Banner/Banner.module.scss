@import '../../styles/config';

.Banner {
  position: relative;
  display: flex;
  padding: spacing();
  margin-bottom: spacing();

  border-radius: border-radius();
  box-shadow: shadow();

  background: color(gray, 10);

  &:before {
    display: block;
    position: absolute;
    content: '';
    top: 0px;
    left: 0px;
    bottom: 0px;

    width: 5px;
    background: color(gray, 6);
    border-radius: border-radius() 0 0 border-radius();
  }
}

.success {
  &:before {
    background: color(green);
  }
}

.info {
  &:before {
    background: color(blue);
  }
}

.warning {
  &:before {
    background: color(yellow, light);
  }
}

.danger {
  &:before {
    background: color(red);
  }
}

.IconWrapper {
  position: relative;
  flex: 1 0 0;
  max-width: 52px;
  margin-right: 18px;
  height: 40px;
}

.Content {
  flex: 1 0 0;
  padding-right: rem(39);

  & > *:last-child {
    margin-bottom: 0;
  }
}

.Children {
  &, & * {
    font-size: font-size(300);
  }
}

.Title {
  margin-bottom: 0;
  font-size: font-size(600);
}

.Actions {
  margin-top: spacing();

  & > * {
    display: inline-block;
    margin-right: spacing();
  }

  & > *:last-child {
    margin-right: 0;
  }
}

.Icon {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  z-index: 1;
}

.successIcon {
  fill: color(green);

  & + .IconBackdrop {
    box-shadow: 0 0 0 7px rgba(color(green), 0.25);
  }
}

.infoIcon {
  fill: darken(color(blue), 8);

  & + .IconBackdrop {
    box-shadow: 0 0 0 7px rgba(color(blue), 0.25);
  }
}

.warningIcon {
  fill: lighten(color(yellow), 4);

  & + .IconBackdrop {
    box-shadow: 0 0 0 7px rgba(color(yellow), 0.25);
  }
}

.dangerIcon {
  fill: color(red);

  & + .IconBackdrop {
    box-shadow: 0 0 0 7px rgba(color(red), 0.25);
  }
}

.IconBackdrop {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background: color(gray, 10);
}

.Dismiss {
  position: absolute;
  right: rem(18);
  top: rem(24);
  border-bottom: none;
  user-select: none;

  &:hover {
    .DismissIcon {
      fill: color(gray, 1);
    }
  }
}

.DismissIcon {
  fill: color(gray, 4);
}
