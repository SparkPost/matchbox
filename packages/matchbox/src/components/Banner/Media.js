import React from 'react';

const Media = React.forwardRef(function Media(props, userRef) {
  return <div ref={userRef}>Media</div>;
});

Media.displayName = 'Banner.Media';

export default Media;
