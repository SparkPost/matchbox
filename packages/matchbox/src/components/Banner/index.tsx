export { default as Banner } from './Banner';
export * from './Banner';
export * from './Action';
export * from './Media';
