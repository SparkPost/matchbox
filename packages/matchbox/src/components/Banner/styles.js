import { tokens } from '@sparkpost/design-tokens';
import { ErrorOutline, CheckCircleOutline, InfoOutline } from '@sparkpost/matchbox-icons';

export function container(props) {
  switch (props.status) {
    case 'success':
      return `
        background: ${tokens.color_green_100};
        border: 1px solid ${tokens.color_green_600};
      `;
    case 'warning':
      return `
        background: ${tokens.color_yellow_100};
        border: 1px solid ${tokens.color_yellow_500};
      `;
    case 'danger':
      return `
        background: ${tokens.color_red_100};
        border: 1px solid ${tokens.color_red_700};
      `;
    // Gray banner no longer exists as of 4.0.0
    case 'default':
    default:
      return `
        background: ${tokens.color_blue_100};
        border: 1px solid ${tokens.color_blue_600};
      `;
  }
}

// Overwrites unstyled link colors within Banner content to be color-contrast accessible
export function childLinks() {
  return `
  p, ul, ol {
    a, a:visited {
      color: ${tokens.color_blue_800};
      &:hover {
        color: ${tokens.color_blue_900};
      }
    }
  }
  `;
}

export const statusIcons = {
  success: {
    iconName: CheckCircleOutline,
    iconLabel: 'Success',
    bg: 'green.600',
    fill: 'white',
  },
  info: {
    iconName: InfoOutline,
    iconLabel: 'Info',
    bg: 'blue.700',
    fill: 'white',
  },
  warning: {
    iconName: ErrorOutline,
    iconLabel: 'Warning',
    bg: 'yellow.300',
    fill: 'yellow.700',
  },
  danger: {
    iconName: ErrorOutline,
    iconLabel: 'Error',
    bg: 'red.700',
    fill: 'white',
  },
};

export function dismissBase() {
  return `
    padding: 0.25rem;
    transition: background ${tokens.motionDuration_fast} ${tokens.motionEase_in_out};
    &:hover {
      cursor: pointer;
    }
  `;
}

export function dismissColor(props) {
  switch (props.status) {
    case 'success':
      return `
        &:hover {
          background: ${tokens.color_green_300};
        }
      `;
    case 'warning':
      return `
        &:hover {
          background: ${tokens.color_yellow_300};
        }
      `;
    case 'danger':
      return `
        &:hover {
          background: ${tokens.color_red_300};
        }
      `;
    // Gray banner no longer exists as of 4.0.0
    case 'default':
    default:
      return `
        &:hover {
          background: ${tokens.color_blue_300};
        }
      `;
  }
}
