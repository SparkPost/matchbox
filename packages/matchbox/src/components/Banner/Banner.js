import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { ErrorIcon, CheckCircle, InfoOutline, Close } from '@sparkpost/matchbox-icons';

import styles from './Banner.module.scss';
import { buttonFrom } from '../Button';

const actionOverrides = { outline: true, size: 'small' };

const IconSection = ({ status }) => {
  const icons = {
    success: CheckCircle,
    info: InfoOutline,
    warning: ErrorIcon,
    danger: ErrorIcon
  };

  if (status === 'default' || !icons[status]) {
    return null;
  }

  const iconClasses = classnames(
    styles.Icon,
    status && styles[`${status}Icon`]
  );

  const Icon = icons[status];

  return (
    <div className={styles.IconWrapper}>
      <Icon size={30} className={iconClasses} />
      <div className={styles.IconBackdrop} />
    </div>
  );
};

class Banner extends Component {
  static displayName = 'Banner';

  static propTypes = {
    /**
     * The type of banner. 'default' | 'success' | 'warning' | 'danger' | 'info'
     */
    status: PropTypes.oneOf(['default', 'success', 'warning', 'danger', 'info']),

    /**
     * The banner's title
     */
    title: PropTypes.string,

    /**
     * Callback when dismiss button is clicked. Button hidden without callback.
     */
    onDismiss: PropTypes.func,

    /**
     * Action that build a button. Most button props will work in here.
     * e.g. { content: 'button label', onClick: callback() }
     */
    action: PropTypes.shape({ content: PropTypes.string.isRequired }),

    /**
     * List of actions that build buttons. Most button props will work in here.
     * Overrides `action`
     */
    actions: PropTypes.arrayOf(PropTypes.shape({
      content: PropTypes.string.isRequired
    })),

    /**
     * Banner Content
     */
    children: PropTypes.node
  };

  static defaultProps = {
    status: 'default'
  };

  render() {
    const {
      children,
      title,
      status,
      action,
      actions,
      onDismiss,
      ...rest
    } = this.props;

    const titleMarkup = title
      ? <h5 className={styles.Title}>{ title }</h5>
      : null;

    let actionMarkup = action
      ? <div className={styles.Actions}>{ buttonFrom(action, actionOverrides) }</div>
      : null;

    if (actions) {
      actionMarkup = (
        <div className={styles.Actions}>
          { actions.map((action, i) => buttonFrom(action, actionOverrides, i)) }
        </div>
      );
    }

    const dismissMarkup = onDismiss
      ? <a className={styles.Dismiss} onClick={onDismiss}><Close size={24} className={styles.DismissIcon} /></a>
      : null;

    const bannerStyles = classnames(
      styles.Banner,
      styles[`${status}`]
    );

    return (
      <div className={bannerStyles} {...rest}>
        <IconSection status={status} />
        <div className={styles.Content}>
          { dismissMarkup }
          { titleMarkup }
          <div className={styles.Children}>{ children }</div>
          { actionMarkup }
        </div>
      </div>
    );
  }
}

export default Banner;
