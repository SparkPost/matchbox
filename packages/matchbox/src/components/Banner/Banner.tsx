import React from 'react';
import PropTypes from 'prop-types';
import { deprecate } from '../../helpers/propTypes';
import { pick, omit } from '../../helpers/props';
import { createPropTypes } from '@styled-system/prop-types';
import { Close } from '@sparkpost/matchbox-icons';
import { Box } from '../Box';
import { Button } from '../Button';
import { Text } from '../Text';
import { Inline } from '../Inline';
import { ScreenReaderOnly } from '../ScreenReaderOnly';
import styled from 'styled-components';
import { container, childLinks, statusIcons, dismissBase, dismissColor } from './styles';
import { buttonReset } from '../../styles/helpers';
import { margin, MarginProps } from 'styled-system';
import { getChild, excludeChild } from '../../helpers/children';
import Action from './Action';
import Media from './Media';

type IconProps = {
  status: 'default' | 'success' | 'warning' | 'danger' | 'info' | 'muted';
  size: 'small' | 'large';
};

function IconSection({ status, size }: IconProps) {
  const statusIcon = React.useMemo(() => {
    return status === 'default' ? statusIcons.info : statusIcons[status];
  }, [statusIcons, status]);

  const Icon = statusIcon.iconName;
  const iconSize = '450';
  const fillColor = statusIcon.fill;

  return (
    <Box
      display="flex"
      flexShrink="0"
      alignItems="center"
      justifyContent="center"
      width={iconSize}
      height={iconSize}
      color={fillColor}
      mr="300"
      mt={size === 'large' ? '4px' : '2px'}
    >
      <Icon size={size === 'large' ? 24 : 20} label={statusIcon.iconLabel} />
    </Box>
  );
}

const StyledContainer = styled(Box)`
  ${container}
  ${margin}
`;

const StyledChildren = styled('div')`
  ${childLinks}
`;

const StyledDismiss = styled(Box)`
  ${buttonReset}
  ${dismissBase}
  ${dismissColor}
`;

type ActionProps = React.ComponentProps<typeof Button> & {
  content: string;
};

interface BannerProps extends React.ComponentPropsWithoutRef<'div'>, MarginProps {
  children?: React.ReactNode;
  'data-id'?: string;
  id?: string;
  title?: string;
  status?: 'default' | 'success' | 'warning' | 'danger' | 'info' | 'muted';
  /**
   * @deprecated Use the Banner.Action component instead
   */
  action?: ActionProps;
  /**
   * @deprecated Use the Banner.Action component instead
   */
  actions?: ActionProps[];

  onDismiss?: (any) => void;
  size?: 'small' | 'large';
}

const Banner = React.forwardRef<HTMLDivElement, BannerProps>(function Banner(
  props: BannerProps,
  userRef,
) {
  const {
    children,
    'data-id': dataId,
    id,
    title,
    status,
    action,
    actions,
    onDismiss,
    size,
    ...rest
  } = props;
  const systemProps = pick(rest, margin.propNames);

  const titleMarkup = title ? (
    <Box mb="200">
      <Text fontSize="400" lineHeight="400" as="h5">
        {title}
      </Text>
    </Box>
  ) : null;

  const actionMarkup = React.useMemo(() => {
    let result = action ? <Action {...omit(action, ['content'])}>{action.content}</Action> : null;

    if (actions) {
      result = (
        <Inline>
          {actions.map((action, i) => (
            <Action key={i} mr="0" {...omit(action, ['content'])}>
              {action.content}
            </Action>
          ))}
        </Inline>
      );
    }

    return result;
  }, [action, actions]);

  const dismissMarkup = onDismiss ? (
    <Box flex={['1', null, '0']} textAlign="right" position="relative">
      <StyledDismiss
        as="button"
        onClick={onDismiss}
        status={status}
        color="gray.800"
        type="button"
        p={size === 'large' ? '100' : 0}
      >
        <ScreenReaderOnly>Dismiss</ScreenReaderOnly>
        <Close size={size === 'large' ? 24 : 20} />
      </StyledDismiss>
    </Box>
  ) : null;

  const hasMedia = React.useMemo(() => {
    return getChild('Banner.Media', children);
  }, [children]);

  const mediaMarkup =
    hasMedia && hasMedia.length ? (
      <Box
        width="100%"
        maxWidth="600px"
        pl="200"
        display={['none', null, 'block']}
        position="relative"
      >
        {getChild('Banner.Media', children)}
      </Box>
    ) : null;

  return (
    <StyledContainer
      display="flex"
      flexWrap={['wrap', null, 'nowrap']}
      p={size === 'large' ? '450' : '300'}
      py={size === 'large' ? null : '200'}
      borderRadius="100"
      status={status}
      {...systemProps}
      ref={userRef}
      tabIndex="-1"
      overflow="hidden"
      data-id={dataId}
      id={id}
    >
      {status !== 'muted' && <IconSection status={status} size={size} />}
      <Box flex="1" order={['1', null, '0']} flexBasis={['100%', null, 'auto']}>
        {titleMarkup}
        <Box>
          <StyledChildren>{excludeChild('Banner.Media', children)}</StyledChildren>
        </Box>
        {actionMarkup}
      </Box>
      {mediaMarkup}
      {dismissMarkup}
    </StyledContainer>
  );
}) as React.ForwardRefExoticComponent<BannerProps> & {
  Media: typeof Media;
  Action: typeof Action;
};

Banner.displayName = 'Banner';
Banner.propTypes = {
  'data-id': PropTypes.string,
  id: PropTypes.string,
  /**
   * The type of banner. 'default' | 'success' | 'warning' | 'danger' | 'info'
   */
  status: PropTypes.oneOf(['default', 'success', 'warning', 'danger', 'info', 'muted']),

  /**
   * The banner's title
   */
  title: PropTypes.string,

  /**
   * Callback when dismiss button is clicked. Button hidden without callback.
   */
  onDismiss: PropTypes.func,

  /**
   * Deprecated in favor of `Banner.Action`
   * Action that build a button. Most button props will work in here.
   * e.g. { content: 'button label', onClick: callback() }
   */
  action: deprecate(
    PropTypes.shape({ content: PropTypes.string.isRequired }),
    'Use `Banner.Action` instead',
  ),

  /**
   * Deprecated in favor of `Banner.Action`
   * List of actions that build buttons. Most button props will work in here.
   * Overrides `action`
   */
  actions: deprecate(
    PropTypes.arrayOf(
      PropTypes.shape({
        content: PropTypes.string.isRequired,
      }),
    ),
    'Use `Banner.Action` instead',
  ),

  /**
   * Banner Content
   */
  children: PropTypes.node,

  /**
   * Banner size
   */
  size: PropTypes.oneOf(['small', 'large']),

  ...createPropTypes(margin.propNames),
};

Banner.defaultProps = {
  status: 'default',
  size: 'large',
};

Banner.Action = Action;
Banner.Media = Media;

export default Banner;
