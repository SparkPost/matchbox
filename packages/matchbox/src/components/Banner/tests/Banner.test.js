import React from 'react';
import Banner from '../Banner';
import { shallow } from 'enzyme';
import styles from '../Banner.module.scss';

describe('Banner', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      title: 'Test Banner',
      onDismiss: jest.fn(),
      action: null,
      actions: null

    };
    wrapper = shallow(<Banner {...props}><p>You know this is a banner</p></Banner>);
  });

  it('renders correctly with default props', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('IconSection').dive().html()).toEqual(null);
  });

  it('renders status', () => {
    wrapper.setProps({ status: 'success' });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('IconSection').dive()).toMatchSnapshot();

    wrapper.setProps({ status: 'danger' });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('IconSection').dive()).toMatchSnapshot();

    wrapper.setProps({ status: 'warning' });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('IconSection').dive()).toMatchSnapshot();

    wrapper.setProps({ status: 'info' });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('IconSection').dive()).toMatchSnapshot();
  });

  it('dismisses banner correctly upon clicking dismiss icon', () => {
    const elem = `a.${styles.Dismiss}`;
    wrapper.find(elem).simulate('click');
    expect(props.onDismiss).toHaveBeenCalledTimes(1);
  });


  it('invokes onClick func of action', () => {
    const action = { content: 'Click me', onClick: jest.fn() };
    wrapper.setProps({ action });
    expect(wrapper).toMatchSnapshot();
    wrapper.find(`.${styles.Actions}`).find('Button').simulate('click');
    expect(action.onClick).toHaveBeenCalledTimes(1);
  });

  it('renders multiple banner actions', () => {
    const actions = [
      { content: 'Click Me 1', onClick: jest.fn() },
      { content: 'Click Me 2', onClick: jest.fn() }
    ];

    wrapper.setProps({ actions });
    expect(wrapper).toMatchSnapshot();
    wrapper.find(`.${styles.Actions}`).find('Button').at(0).simulate('click');
    expect(actions[0].onClick).toHaveBeenCalledTimes(1);

    wrapper.find(`.${styles.Actions}`).find('Button').at(1).simulate('click');
    expect(actions[1].onClick).toHaveBeenCalledTimes(1);
  });
});
