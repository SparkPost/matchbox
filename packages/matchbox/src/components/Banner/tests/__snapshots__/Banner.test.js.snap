// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner invokes onClick func of action 1`] = `
<div
  className="Banner default"
>
  <IconSection
    status="default"
  />
  <div
    className="Content"
  >
    <a
      className="Dismiss"
      onClick={[MockFunction]}
    >
      <Close
        className="DismissIcon"
        size={24}
      />
    </a>
    <h5
      className="Title"
    >
      Test Banner
    </h5>
    <div
      className="Children"
    >
      <p>
        You know this is a banner
      </p>
    </div>
    <div
      className="Actions"
    >
      <Button
        onClick={[MockFunction]}
        outline={true}
        size="small"
      >
        Click me
      </Button>
    </div>
  </div>
</div>
`;

exports[`Banner renders correctly with default props 1`] = `
<div
  className="Banner default"
>
  <IconSection
    status="default"
  />
  <div
    className="Content"
  >
    <a
      className="Dismiss"
      onClick={[MockFunction]}
    >
      <Close
        className="DismissIcon"
        size={24}
      />
    </a>
    <h5
      className="Title"
    >
      Test Banner
    </h5>
    <div
      className="Children"
    >
      <p>
        You know this is a banner
      </p>
    </div>
  </div>
</div>
`;

exports[`Banner renders multiple banner actions 1`] = `
<div
  className="Banner default"
>
  <IconSection
    status="default"
  />
  <div
    className="Content"
  >
    <a
      className="Dismiss"
      onClick={[MockFunction]}
    >
      <Close
        className="DismissIcon"
        size={24}
      />
    </a>
    <h5
      className="Title"
    >
      Test Banner
    </h5>
    <div
      className="Children"
    >
      <p>
        You know this is a banner
      </p>
    </div>
    <div
      className="Actions"
    >
      <Button
        key="0"
        onClick={[MockFunction]}
        outline={true}
        size="small"
      >
        Click Me 1
      </Button>
      <Button
        key="1"
        onClick={[MockFunction]}
        outline={true}
        size="small"
      >
        Click Me 2
      </Button>
    </div>
  </div>
</div>
`;

exports[`Banner renders status 1`] = `
<div
  className="Banner success"
>
  <IconSection
    status="success"
  />
  <div
    className="Content"
  >
    <a
      className="Dismiss"
      onClick={[MockFunction]}
    >
      <Close
        className="DismissIcon"
        size={24}
      />
    </a>
    <h5
      className="Title"
    >
      Test Banner
    </h5>
    <div
      className="Children"
    >
      <p>
        You know this is a banner
      </p>
    </div>
  </div>
</div>
`;

exports[`Banner renders status 2`] = `
<div
  className="IconWrapper"
>
  <CheckCircle
    className="Icon successIcon"
    label="Success"
    size={30}
  />
  <div
    className="IconBackdrop"
  />
</div>
`;

exports[`Banner renders status 3`] = `
<div
  className="Banner danger"
>
  <IconSection
    status="danger"
  />
  <div
    className="Content"
  >
    <a
      className="Dismiss"
      onClick={[MockFunction]}
    >
      <Close
        className="DismissIcon"
        size={24}
      />
    </a>
    <h5
      className="Title"
    >
      Test Banner
    </h5>
    <div
      className="Children"
    >
      <p>
        You know this is a banner
      </p>
    </div>
  </div>
</div>
`;

exports[`Banner renders status 4`] = `
<div
  className="IconWrapper"
>
  <Error
    className="Icon dangerIcon"
    label="Error"
    size={30}
  />
  <div
    className="IconBackdrop"
  />
</div>
`;

exports[`Banner renders status 5`] = `
<div
  className="Banner warning"
>
  <IconSection
    status="warning"
  />
  <div
    className="Content"
  >
    <a
      className="Dismiss"
      onClick={[MockFunction]}
    >
      <Close
        className="DismissIcon"
        size={24}
      />
    </a>
    <h5
      className="Title"
    >
      Test Banner
    </h5>
    <div
      className="Children"
    >
      <p>
        You know this is a banner
      </p>
    </div>
  </div>
</div>
`;

exports[`Banner renders status 6`] = `
<div
  className="IconWrapper"
>
  <Error
    className="Icon warningIcon"
    label="Warning"
    size={30}
  />
  <div
    className="IconBackdrop"
  />
</div>
`;

exports[`Banner renders status 7`] = `
<div
  className="Banner info"
>
  <IconSection
    status="info"
  />
  <div
    className="Content"
  >
    <a
      className="Dismiss"
      onClick={[MockFunction]}
    >
      <Close
        className="DismissIcon"
        size={24}
      />
    </a>
    <h5
      className="Title"
    >
      Test Banner
    </h5>
    <div
      className="Children"
    >
      <p>
        You know this is a banner
      </p>
    </div>
  </div>
</div>
`;

exports[`Banner renders status 8`] = `
<div
  className="IconWrapper"
>
  <InfoOutline
    className="Icon infoIcon"
    label="Info"
    size={30}
  />
  <div
    className="IconBackdrop"
  />
</div>
`;
